packages/nuxt/src/core/templates.ts

```diff
@@ -103,7 +103,7 @@ export const pluginsDeclaration: NuxtTemplate<TemplateContext> = {
   getContents: (ctx) => {
     const EXTENSION_RE = new RegExp(`(?<=\\w)(${ctx.nuxt.options.extensions.map(e => escapeRE(e)).join('|')})$`, 'g')
     const tsImports = ctx.app.plugins
-      .filter(p => !isAbsolute(p.src) || existsSync(p.src) || existsSync(p.src.replace(EXTENSION_RE, '.d.ts')))
+      .filter(p => !isAbsolute(p.src) || existsSync(p.src) || (p.src.includes('.d.ts') && existsSync(p.src.replace(EXTENSION_RE, '.d.ts'))))
```

Comment:
this doesn't look quite right to me...

can you provide an example?

Reply:
Do you have any suggestions, please?

Here is a branch with a plugin example in the playground : 

https://github.com/nasrob/nuxt/tree/fix/22617-demo/playground/plugins

It prints the message using the plugin, as shown in the screenshot. However, I get this message
`'__VLS_ctx.$hello' is of type 'unknown'.ts(18046)` in the `template` of `app.vue`

![Capture d’écran 2023-10-26 à 23 29 45](https://github.com/nuxt/nuxt/assets/11391766/ae5aaa96-7993-4666-838c-8626e9ec26b0)